custom/abt/reports/views.py

```diff
@@ -117,3 +124,130 @@ def email_response(self):
         return HttpResponse(json.dumps({
             'report': '',
         }))
+
+
+class UniqueSOPSumDataSource(SqlData):
+
+    def __init__(self, domain, filter_values):
+        config = {
+            'domain': domain,
+        }
+        for (key, value) in filter_values.items():
+            if key == 'date_of_data_collection' and value:
+                config['start_date'] = value.startdate
+                config['end_date'] = value.enddate
+            elif key and value:
+                self.__setattr__(key, [x.value for x in value])
+                config[key] = self.__getattribute__(key)
+                clean_IN_filter_value(config, key)
+        self.config = config
+
+    @property
+    def table_name(self):
+        return get_table_name(self.config['domain'], "static-sms-indicators-001")
+
+    @property
+    def filters(self):
+        filters = []
+        if 'start_date' in self.config and 'end_date' in self.config:
+            filters.append(BETWEEN("date_of_data_collection", "start_date", "end_date"))
+        if 'country' in self.config:
+            filters.append(IN('country', get_INFilter_bindparams('country', self.__getattribute__("country"))))
+        if 'level_1' in self.config:
+            filters.append(IN('level_1', get_INFilter_bindparams('level_1', self.__getattribute__("level_1"))))
+        if 'level_2' in self.config:
+            filters.append(IN('level_2', get_INFilter_bindparams('level_2', self.__getattribute__("level_2"))))
+        if 'level_3' in self.config:
+            filters.append(IN('level_3', get_INFilter_bindparams('level_3', self.__getattribute__("level_3"))))
+        if 'level_4' in self.config:
+            filters.append(IN('level_4', get_INFilter_bindparams('level_4', self.__getattribute__("level_4"))))
+        return filters
+
+    @property
+    def group_by(self):
+        return ['country', 'level_1', 'level_2', 'level_3', 'level_4']
+
+    @property
+    def columns(self):
+        return [
+            DatabaseColumn('country', SimpleColumn('country')),
+            DatabaseColumn('level_1', SimpleColumn('level_1')),
+            DatabaseColumn('level_2', SimpleColumn('level_2')),
+            DatabaseColumn('level_3', SimpleColumn('level_3')),
+            DatabaseColumn('level_4', SimpleColumn('level_4')),
+            DatabaseColumn('total_sprayers', MaxColumn('total_sprayers'))
+        ]
+
+
+class CombinedDataSource(object):
+
+    def __init__(self, original_data_source, filter_values):
+        self.original_data_source = original_data_source
+        self.filters_data = filter_values
+
+    def get_data(self, start, limit):
+        original_data = self.original_data_source.get_data(start, limit)
+        unique_sops = UniqueSOPSumDataSource(self.original_data_source.domain, self.filters_data).get_data()
```

Comment:
I noticed that the SOP value stays the same regardless of the date range you select in https://www.commcarehq.org/a/vectorlink-burkina-faso/reports/custom_configurable/custom-vectorlink-burkina-faso-spray-progress-report-level-1/. I wonder if something about how the filters are being passed in is dropping the filter input somehow?

Reply:
Oh okay never mind, I just saw https://github.com/dimagi/commcare-hq/pull/22207